lib/Parse/ParseExpr.cpp

```diff
@@ -422,6 +422,20 @@ ParserResult<Expr> Parser::parseExprSequenceElement(Diag<> message,
     return sub;
   }
 
+  if (Tok.isContextualKeyword("copy") &&
+      peekToken().isAny(tok::identifier, tok::kw_self) &&
```

Comment:
As I discussed offline with @gottesmm, I think this also needs to include dollar identifiers for anonymous closure arguments.

Reply:
I fixed that.